### PR TITLE
Display model diagram for h5mlm datatype

### PR DESCRIPTION
**Pull request by qiagu**
(Please replace this header with a description of your pull request. Please include *BOTH* what you did and why you made the changes. The "why" may simply be citing a relevant Galaxy issue.)
(If fixing a bug, please add any relevant error or traceback)
(For UI components, it is recommended to include screenshots or screencasts)

## How to test the changes?
(Select all options that apply)
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

**Comment by qiagu** [comment]
image

File: lib/galaxy/datatypes/binary.py [hunk 1]
```diff
@@ -1869,6 +1869,12 @@ class H5MLM(H5):
     max_peek_size = 1000  # 1 KB
     max_preview_size = 1000000  # 1 MB
 
+    _CONFIG = '-model_config-'
+    _HTTP_REPR = '-http_repr-'
+    _HYPERPARAMETER = '-model_hyperparameters-'
+    _REPR = '-repr-'
+    _URL = '-URL-'
+
```
Comment (jdavcs) [inline lib/galaxy/datatypes/binary.py]: Very minor, but do we need these leading underscores? That makes it a "private constant", but why mark it as private if it's a constant? Or is there some other meaning implied here?

File: lib/galaxy/datatypes/binary.py [hunk 2]
```diff
@@ -1900,28 +1906,43 @@ def set_meta(self, dataset, overwrite=True, metadata_tmp_files_dir=None, **kwd):
...
+                repr_ = util.unicodify(handle.attrs.get(self._REPR))
+            if len(repr_) <= self.max_preview_size:
+                return repr_
+            else:
+                return "<p><strong>The model representation is too big to be displayed!</strong></p>"
         except Exception as e:
             log.warning("%s, get_repr Except: %s", self, e)
             return ""
 
+    def get_html_repr(self, filename):
```
Comment (jdavcs) [inline lib/galaxy/datatypes/binary.py]: If you don't mind, could you factor out the common logic from these 2 methods into a helper method? That way you could get rid of 10 lines of (almost) duplicate code. Something like this, maybe:
```
def get_repr:
    return repr_helper(repr args)

def get_html_repr:
    return repr_helper(html-repr args)

def repr_helper(args):
    main logic, parametrized with args
```

File: lib/galaxy/datatypes/binary.py (hunk 1 above)
Comment (qiagu) [reply lib/galaxy/datatypes/binary.py]: Updated. Thanks!

File: lib/galaxy/datatypes/binary.py (hunk 2 above)
Comment (qiagu) [reply lib/galaxy/datatypes/binary.py]: Nice point! How about now?

Comment (jdavcs) [reply lib/galaxy/datatypes/binary.py]: Looks good, thank you!

**Review by jdavcs — APPROVED** [review]
Thank you for addressing the suggestions!

**Comment by jdavcs** [comment]
@qiagu a rebase onto dev should fix this.

**Comment by jdavcs** [comment]
@qiagu I've rebased and pushed to your branch (hope you don't mind). This should resolve the issues. I'll merge this when  the tests pass.

**Comment by qiagu** [comment]
@jdavcs Thanks you for taking care of this!

**Comment by jdavcs** [comment]
@qiagu can you, please, resolve the conflicts next week? Conflicts are in `binary.py` (triggered by merging #15052). It's new type hints + renaming of a variable: it used to be `rval`, which switched its type inside the method body; to accommodate proper type-checking it was split into `rval_dict` and `rval`,  so you could do the same with your `out` variable. Ping me if something doesn't make sense.

**Comment by jdavcs** [comment]
Test failures are unrelated. Thank you for resolving the conflicts, @qiagu!

**Comment by qiagu** [comment]
Thanks @jdavcs

**Pull request merged**